src/test/java/hudson/plugins/mercurial/MercurialSCMSourceTest.java

```diff
@@ -0,0 +1,74 @@
+package hudson.plugins.mercurial;
+
+import hudson.AbortException;
+import hudson.tools.ToolProperty;
+import hudson.util.LogTaskListener;
+import hudson.util.StreamTaskListener;
+import jenkins.scm.api.SCMRevision;
+import org.junit.*;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.util.Collections;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class MercurialSCMSourceTest {
+
+    @ClassRule public static JenkinsRule r = new JenkinsRule();
+    @ClassRule public static MercurialSampleRepoRule sampleRepo = new MercurialSampleRepoRule();
+
+    private static LogTaskListener listener;
+    private static MercurialSCMSource mercurialSCMSource;
+
+    @BeforeClass
+    public static void prepareEnvironment() throws Exception {
+        String instName = "caching";
+        MercurialInstallation installation = new MercurialInstallation(instName, "", "hg", false, true, false, null,
+                Collections.<ToolProperty<?>>emptyList());
+        listener = new LogTaskListener(Logger.getLogger(MercurialSCMSourceTest.class.getName()), Level.INFO);
+        r.jenkins.getDescriptorByType(MercurialInstallation.DescriptorImpl.class).setInstallations(installation);
+        installation.forNode(r.jenkins, StreamTaskListener.fromStdout());
+        sampleRepo.init();
+        sampleRepo.write("file", "initial content");
+        sampleRepo.hg("commit", "--addremove", "--message=initial");
+        sampleRepo.hg("tag", "version-1.0");
+        sampleRepo.hg("branch", "my-branch");
+        sampleRepo.write("file2", "content in branch");
+        sampleRepo.hg("commit", "--addremove", "--message=branch");
+        sampleRepo.hg("tag", "version-1.1");
+
+
+        installation.forNode(r.jenkins, StreamTaskListener.fromStdout());
+        mercurialSCMSource = new MercurialSCMSource(null, instName, sampleRepo.fileUrl(), null, null, null, null, null, true);
+    }
+
+    @Test public void testRetrieveUnknownRevision() throws Exception {
+        try {
+            mercurialSCMSource.retrieve("does_not_exist", listener);
+        } catch (AbortException e) {
+            return; // expected
```

Comment:
I would have expected this to return null, as in the super Javadoc.

This could be accomplished most easily by using `log` with the revset

    present(…)

and checking for empty output.